flink-datastream/src/main/java/org/apache/flink/datastream/impl/stream/NonKeyedPartitionStreamImpl.java

```diff
@@ -73,9 +80,27 @@ public <OUT> ProcessConfigurableAndNonKeyedPartitionStream<OUT> process(
 
         TypeInformation<OUT> outType =
                 StreamUtils.getOutputTypeForOneInputProcessFunction(processFunction, getType());
-        ProcessOperator<T, OUT> operator = new ProcessOperator<>(processFunction);
-        OneInputTransformation<T, OUT> outputTransform =
-                StreamUtils.getOneInputTransformation("Process", this, outType, operator);
+        OneInputTransformation<T, OUT> outputTransform;
```

Comment:
I think for methods which doesn't support window, for example:

```public <T_OTHER, OUT> ProcessConfigurableAndNonKeyedPartitionStream<OUT> connectAndProcess(
            BroadcastStream<T_OTHER> other,
            TwoInputBroadcastStreamProcessFunction<T, T_OTHER, OUT> processFunction)
```

we should check the `TwoInputBroadcastStreamProcessFunction` is not `InternalTwoInputNonBroadcastWindowStreamProcessFunction`, otherwise it may produce unexpected behavior as it is a valid `TwoInputNonBroadcastStreamProcessFunction` but have dummy `processRecord` impl.

Please also check this for other stream.